### PR TITLE
add: `mcontrolcenter-bin`

### PR DESCRIPTION
**Pull request by soumyatheman**

### CODE REVIEW

**Review by wizard-28 — CHANGES_REQUESTED** [review]
Run pre-commit hooks...

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	read ans
+	if [ ${ans} == y ] || [ ${ans} == Y ]; then
+		cp /usr/share/applications/mcontrolcenter.desktop ~/.config/autostart/mcontrolcenter.desktop
+		echo "Copied desktop file to ~/.config/autostart folder, programme will now start upon startup."
+	elif [ ${ans} == n ] || [ ${ans} == N ]; then
+		echo "Programme will not start upon startup"
+	else
+		echo "Interpreting as Y"
+		cp /usr/share/applications/mcontrolcenter.desktop ~/.config/autostart/mcontrolcenter.desktop
+	fi
```
Comment (wizard-28) [inline packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:31]: Use the `ask` function for this. Example: https://github.com/pacstall/pacstall-programs/blob/35a956766dba8f325b24f801d80ac9733efcba70/packages/librewolf-app/librewolf-app.pacscript#L42-L44

https://github.com/pacstall/pacstall-programs/blob/35a956766dba8f325b24f801d80ac9733efcba70/packages/tuner-git/tuner-git.pacscript#L13-L17

---

@Henryws, I don't think `fancy_message` and `ask` are documented in the wiki?

Comment (soumyatheman) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:31]: Those aren't really documented

Comment (Henryws) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:31]: Added at the bottom of the Pacscript 101 page.

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	sudo install -Dm755 ${gives} "$pkgdir"/usr/bin/${gives}
+	sudo install -Dm644 ${gives}.desktop $pkgdir/usr/share/applications/${gives}.desktop
+	sudo install -Dm644 ${gives}.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/${gives}.svg
+	sudo install -Dm755 ${gives}-helper "$pkgdir"/usr/libexec/${gives}-helper
+	sudo install -Dm644 ${gives}-helper.conf "$pkgdir"/usr/share/dbus-1/system.d/${gives}-helper.conf
+	sudo install -Dm644 ${gives}.helper.service "$pkgdir"/usr/share/dbus-1/system-services/${gives}.helper.service
+}
+postinst() {
+	echo "To start this programme of startup, type:
+	cp /usr/share/applications/mcontrolcenter.desktop ~/.config/autostart/mcontrolcenter.desktop"
```
Comment (wizard-28) [inline packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]: Why did you remove the prompt?

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+}
```
Comment (wizard-28) [inline packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:23]: Add a `removescript` removing this file on uninstallation.

**Review by wizard-28 — CHANGES_REQUESTED** [review]
Run pre-commit hooks.

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript (hunk 3 above)
Comment (soumyatheman) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:23]: This is a manual process, so no need of that.

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript (hunk 2 above)
Comment (soumyatheman) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]: Because doing so, if one doesn't have a autostart folder, creating it from the `postinst` hook will make it under root user, also there'll be permission issues later on, as other apps also use that directory. I don't think interfering into someone's home directory, as a root user a good thing. Those apps which need the directory, often make them during the first run.

Comment (wizard-28) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]: You can use `sudo -u ${PACSTALL_USER} <cmd>` to de-escalate your privileges.

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript (hunk 3 above)
Comment (wizard-28) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:23]: You can still remove that file location with `rm -f` regardless of if the user decides to install the autostart file. This way, no junk files are left behind on the system on uninstallation.

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript (hunk 2 above)
Comment (Henryws) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]: You should probably add a prompt to ask if they want to install, *however*, print out the command that will run before.

**Review by wizard-28 — CHANGES_REQUESTED** [review]

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+removescript() {
+	rm -rf ~/.config/autostart/mcontrolcenter.desktop
```
[the remark this answers is not in the document]
Comment (Henryws) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:25]: https://github.com/pacstall/pacstall/wiki/Pacscript-101#other

Added.

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript (hunk 2 above)
Comment (soumyatheman) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]: > You can use `sudo -u ${PACSTALL_USER} <cmd>` to de-escalate your privileges.

No help
`````
Do you want this programme to start at startup? [Y/n] y
cp: cannot create regular file '/home/soumyaxubuntu/.config/autostart/mcontrolcenter.desktop': Permission denied

Comment (soumyatheman) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]: The code
````
postinst() {
	if ask "Do you want this programme to start at startup?" Y; then
	sudo -u ${PACSTALL_USER} cp /usr/share/applications/mcontrolcenter.desktop ~/.config/autostart/mcontrolcenter.desktop
	echo "Copied desktop file to ~/.config/autostart folder, programme will now start upon startup."
	else
		echo "Programme will not start upon startup"
	fi
}

Comment (Henryws) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]: Ok, so create the directory before copying.

Comment (soumyatheman) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]: I already have it anyways

Comment (Henryws) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]: Have what?

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	sudo install -Dm755 ${gives} "$pkgdir"/usr/bin/${gives}
+	sudo install -Dm644 ${gives}.desktop $pkgdir/usr/share/applications/${gives}.desktop
+	sudo install -Dm644 ${gives}.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/${gives}.svg
+	sudo install -Dm755 ${gives}-helper "$pkgdir"/usr/libexec/${gives}-helper
+	sudo install -Dm644 ${gives}-helper.conf "$pkgdir"/usr/share/dbus-1/system.d/${gives}-helper.conf
+	sudo install -Dm644 ${gives}.helper.service "$pkgdir"/usr/share/dbus-1/system-services/${gives}.helper.service
+}
+postinst() {
+	if ask "Do you want this programme to start at startup?" Y; then
+  sudo -u ${PACSTALL_USER} mkdir ${homedir}/.config/autostart
```
Comment (Henryws) [inline packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]:
```suggestion
  sudo -u ${PACSTALL_USER} mkdir -p "${homedir}/.config/autostart/"
```

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript (hunk 5 above)
Comment (soumyatheman) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:22]: A .config folder is anways gonna exist

Comment (Henryws) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript]: Maybe not autostart. It's better to just make sure it exists rather than not.

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+maintainer="সৌম্যদীপ <soumyadeepghosh2004@zohomail.in>"
+
+install() {
+	cd app
+	sudo install -Dm755 ${gives} "$pkgdir"/usr/bin/${gives}
+	sudo install -Dm644 ${gives}.desktop $pkgdir/usr/share/applications/${gives}.desktop
+	sudo install -Dm644 ${gives}.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/${gives}.svg
+	sudo install -Dm755 ${gives}-helper "$pkgdir"/usr/libexec/${gives}-helper
+	sudo install -Dm644 ${gives}-helper.conf "$pkgdir"/usr/share/dbus-1/system.d/${gives}-helper.conf
+	sudo install -Dm644 ${gives}.helper.service "$pkgdir"/usr/share/dbus-1/system-services/${gives}.helper.service
```
Comment (Henryws) [inline packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:18]: Quote these as pre-commit would.

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+depends="libqt5widgets5"
+url="https://github.com/dmitry-s93/MControlCenter/releases/download/${version}/MControlCenter-${version}.tar.gz"
+repology=("project: mcontrolcenter")
+description="An application that allows you to change the settings of MSI laptops."
+hash="bab563f692fb3d42494d95de21611d230e0b0736e2e80f08cf8a1b7b990d9f0f"
+maintainer="সৌম্যদীপ <soumyadeepghosh2004@zohomail.in>"
+
+install() {
+	cd app
+	sudo install -Dm755 ${gives} "$pkgdir"/usr/bin/${gives}
```
Comment (Henryws) [inline packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:13]: Variables are still not quoted.

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+	sudo install -Dm644 "${gives}-helper.conf" "$pkgdir"/usr/share/dbus-1/system.d/"${gives}-helper.conf"
+	sudo install -Dm644 "${gives}.helper.service" "$pkgdir"/usr/share/dbus-1/system-services/"${gives}.helper.service"
+}
+postinst() {
+	if ask "Do you want this programme to start at startup?" Y; then
+    sudo -u "${PACSTALL_USER}" mkdir -p "${homedir}"/.config/autostart
+		sudo -u "${PACSTALL_USER}" cp /usr/share/applications/mcontrolcenter.desktop "${homedir}"/.config/autostart/mcontrolcenter.desktop
+		echo "Copied desktop file to ${homedir}/.config/autostart folder, programme will now start upon startup."
+	else
+		echo "Programme will not start upon startup"
```
Comment (wizard-28) [inline packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:26]:
```suggestion
		fancy_message info "Programme will not autostart"
```

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+	cd app
+	sudo install -Dm755 "${gives}" "$pkgdir"/usr/bin/"${gives}"
+	sudo install -Dm644 "${gives}.desktop" "$pkgdir"/usr/share/applications/"${gives}.desktop"
+	sudo install -Dm644 "${gives}.svg" "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"${gives}.svg"
+	sudo install -Dm755 "${gives}-helper" "$pkgdir"/usr/libexec/"${gives}-helper"
+	sudo install -Dm644 "${gives}-helper.conf" "$pkgdir"/usr/share/dbus-1/system.d/"${gives}-helper.conf"
+	sudo install -Dm644 "${gives}.helper.service" "$pkgdir"/usr/share/dbus-1/system-services/"${gives}.helper.service"
+}
+postinst() {
+	if ask "Do you want this programme to start at startup?" Y; then
```
Comment (wizard-28) [inline packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:21]:
```suggestion
	if ask "Do you want this programme to autostart?" Y; then
```

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    sudo -u "${PACSTALL_USER}" mkdir -p "${homedir}"/.config/autostart
+		sudo -u "${PACSTALL_USER}" cp /usr/share/applications/mcontrolcenter.desktop "${homedir}"/.config/autostart/mcontrolcenter.desktop
+		echo "Copied desktop file to ${homedir}/.config/autostart folder, programme will now start upon startup."
```
Comment (wizard-28) [inline packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:24]:
```suggestion
		fancy_message info "Copied desktop file to ${homedir}/.config/autostart folder, programme will now start upon startup."
```

**Review by wizard-28 — CHANGES_REQUESTED** [review]
Run the pre-commit hooks.........

---

Please don't use GitHub's online file editor, and actually commit and push the changes from your local development environment, so that the hooks run automatically (Install the hooks again if you've deleted your cloned fork).

File: packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript (hunk 10 above)
Comment (soumyatheman) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:24]: Draft out this PR.

Comment (Henryws) [reply packages/mcontrolcenter-bin/mcontrolcenter-bin.pacscript:24]: Can you not draft it yourself?
